src/cpr_data_access/models.py

```diff
@@ -771,6 +822,118 @@ def add_spans(
 
         return self
 
+    def add_metadata(
+        self,
+        target_model: type[AnyDocument],
+        metadata_csv_path: Path,
+        force_all_documents_have_metadata: bool = True,
+    ) -> "Dataset":
+        """
+        Convert all documents in the dataset to the target model, by adding metadata from the metadata CSV.
+
+        :param target_model: model to convert documents in dataset to
+        :param metadata_csv_path: path to metadata CSV
+        :param force_all_documents_have_metadata: whether to raise an error if any documents in the dataset do not have metadata in the CSV. **Warning: if set to false, the output dataset may have fewer documents than before running this function.** Defaults to True
```

Comment:
Surely over 89 characters haha?

Reply:
the pre-commit hooks don't enforce this hence CI passing (i disabled that check because i find it annoying 😉)